### ISSUE TITLE
Stripping comments fails for last line

### ISSUE THREAD

**Issue by GerardAtBottomLine** [mikefarah/yq#1321]
**Describe the bug**
I run into an issue when [stripping comments](https://mikefarah.gitbook.io/yq/operators/comment-operators#remove-strip-all-comments): Stripping comments appears to fail when the comment is the only contents of the last line of the file.

Version of yq: 4.26.1
Operating system: linux
Installed via: docker

**Input Yaml**
data1.yml:
```yaml
a: cat # comment
# great
b: # key comment
# last line is a comment
```
**Command**
```
docker run --rm --volume "$(pwd)":/workdir mikefarah/yq:4 eval-all '... comments=""' data1.yml
```

**Actual behavior**

```yaml
a: cat
b:
# last line is a comment
```

**Expected behavior**

```yaml
a: cat
b:
```

**Comment by mikefarah** [comment]
Yep that's a bug 👍🏼  will fix in the next release.

**Comment by mikefarah** [comment]
Fixed in 4.27.5

**Comment by GerardAtBottomLine** [comment]
Thanks!
